### PR TITLE
Complete moxfield service integration

### PR DESCRIPTION
**Pull request by AxiomaticAdi**

### CODE REVIEW

**Comment by AxiomaticAdi** [comment]
Proof code is working

image

File: src/components/playerOverview/playerDetails/PlayerDetailsInfoCard.tsx [hunk 1]
```diff
@@ -23,6 +28,19 @@ export const PlayerDetailsInfoCard = React.memo(function PlayerDetailsInfoCard({
     const profileId = potentialProfileId ?? "";
     const profile = useSelector((state: AppState) => ProfileSelectors.getProfile(state, profileId));
 
+    const moxfieldProfile: MoxfieldProfile | undefined = useSelector((state: AppState) => {
+        if (state.profiles.moxfieldProfiles !== undefined && profile?.moxfieldId !== undefined) {
```
Comment (davidpchi) [inline src/components/playerOverview/playerDetails/PlayerDetailsInfoCard.tsx:32]: your next task is to make this into a real selector. we can table this for now though

File: src/components/playerOverview/playerDetails/PlayerDetailsInfoCard.tsx [hunk 2]
```diff
@@ -23,6 +28,19 @@ export const PlayerDetailsInfoCard = React.memo(function PlayerDetailsInfoCard({
...
+            return state.profiles.moxfieldProfiles[profile.moxfieldId];
+        }
+        return undefined;
+    });
+
+    useEffect(() => {
```
Comment (davidpchi) [inline src/components/playerOverview/playerDetails/PlayerDetailsInfoCard.tsx:38]: nit: can we add a comment here explaining what this useEffect is doing? makes it clearer what the intention here is (when do we hydrate?)

File: src/redux/profiles/profilesActions.ts [hunk 3]
```diff
@@ -1,13 +1,23 @@
...
+    HydrateMoxfieldProfileComplete = "ProfileActions/HydrateMoxfieldProfileComplete"
 }
 
 export const ProfilesAction = {
     GetProfilesComplete: createAction(ProfilesActionType.GetProfilesComplete, (data: Profile[]) => ({
         type: ProfilesActionType.GetProfilesComplete,
         payload: data
-    }))
+    })),
+
```
Comment (davidpchi) [inline src/redux/profiles/profilesActions.ts:15]: nit: remove space

File: src/redux/profiles/profilesReducer.ts [hunk 4]
```diff
@@ -11,18 +12,31 @@ export type ProfilesState = Readonly<{
      * A map of all profiles where the ID is the discord id.
      */
     profiles: { [id: string]: Profile } | undefined;
+    moxfieldProfiles: { [id: string]: MoxfieldProfile } | undefined;
```
Comment (davidpchi) [inline src/redux/profiles/profilesReducer.ts:15]: nit: can we add JS Doc

File: src/redux/profiles/profilesReducer.ts [hunk 5]
```diff
@@ -11,18 +12,31 @@ export type ProfilesState = Readonly<{
...
-    });
+    builder
+        .addCase(ProfilesAction.GetProfilesComplete, (state, action) => {
+            const result: { [id: string]: Profile } = {};
+            for (const item of action.payload) {
+                result[item.id] = item;
+            }
+            state.profiles = result;
+        })
+
```
Comment (davidpchi) [inline src/redux/profiles/profilesReducer.ts:32]: nit: remove space

File: src/services/MoxfieldService.ts [hunk 6]
```diff
@@ -24,6 +26,31 @@ const useGetMoxfieldProfile = () => {
...
+                    headers: { "Content-Type": "application/json" }
+                });
+                const serviceObj: MoxfieldResponseData = _res.data;
+                let newMoxfieldProfile: MoxfieldProfile = {
+                    userName: serviceObj.userName,
+                    imageUrl: serviceObj.profileImageUrl ? serviceObj.profileImageUrl : undefined
+                };
+                dispatch(ProfilesAction.HydrateMoxfieldProfileComplete(newMoxfieldProfile));
+            } catch (err) {
+                console.log(err);
```
Comment (davidpchi) [inline src/services/MoxfieldService.ts:46]: console.error, and probably a better log message than this

**Review by davidpchi — CHANGES_REQUESTED** [review]

**Review by davidpchi — APPROVED** [review]

**Pull request merged**